internal/internal_worker_base.go

```diff
@@ -246,6 +247,18 @@ func (bw *baseWorker) pollTask() {
 }
 
 func (bw *baseWorker) processTask(task interface{}) {
+	defer func() {
+		if p := recover(); p != nil {
+			bw.metricsScope.Counter(metrics.WorkerPanicCounter).Inc(1)
```

Comment:
could we call fail task from here?

Reply:
We can not do it here, but we could put panic handler one layer lower to the activityPoller and workflowPoller.